host/windows/syscall.c

```diff
@@ -324,6 +324,292 @@ __declspec(noreturn) static void _panic(
 **==============================================================================
 */
 
+// Allocates char* string which follows the expected rules for
+// enclaves. Paths in the format
+// <driveletter>:\<item>\<item> -> /<driveletter>/<item>/item>
+// <driveletter>:/<item>/<item> -> /<driveletter>/<item>/item>
+// paths without drive letter are detected and the drive added
+// /<item>/<item> -> /<current driveletter>/<item>/item>
+// relative paths are translated to absolute with drive letter
+// returns null if the string is illegal
+//
+// The string  must be freed
+//
+// we don't handle paths which start with the "\\?\" thing. Since we convert
+// everything to wchar, we have normal paths up to 32767 chars long. We never
+// use the 8 bit version of the win32 apis. That said, we get paths in 8 bit
+// characters from the enclave because there is no "wopen" in linux
+//
+// We always convert in the ocall function.
+//
+char* oe_win_path_to_posix(const char* path)
+{
+    size_t required_size = 0;
+    size_t current_dir_len = 0;
+    char* current_dir = NULL;
+    char* enclave_path = NULL;
+
+    if (!path)
+    {
+        return NULL;
+    }
+    // Relative or incomplete path?
+
+    // absolute path with drive letter.
+    // we do not handle device type paths ("CON:) or double-letter paths in case
+    // of really large numbers of disks (>26). If you have those, mount on
+    // windows
+    //
+    if (isalpha(path[0]) && path[1] == ':')
+    {
+        // Abosolute path is drive letter
+        required_size = strlen(path) + 1;
+    }
+    else if (path[0] == '/' || path[0] == '\\')
+    {
+        required_size = strlen(path) + 3; // Add a drive letter to the path
+    }
+    else
+    {
+        current_dir = _getcwd(NULL, 32767);
```

Comment:
Is there a better way to do this than allocating 32K buffer every time?

Reply:
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/getcwd-wgetcwd?view=vs-2019
   char* buffer;

   // Get the current working directory:
   if ( (buffer = _getcwd( NULL, 0 )) == NULL )
      perror( "_getcwd error" );

Reply:
@nonpolarity I think you can make the second argument 0, based on https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/getcwd-wgetcwd?view=vs-2019

Reply:
@radhikaj Beautiful solution! I thought I need the max path length 260.